build_mt.py

```diff
@@ -1,599 +1,827 @@
+#!/usr/bin/env python3
+'''
+This is the first script in the "lightcone halo" pipeline.  The goal of this script is to use merger
+tree information to flag halos that intersect the lightcone and make a unique determination of which
+halo catalog epoch from which to draw the halo.
+
+Usage
+-----
+$ ./build_mt.py --help
+'''
+
+import sys
 import glob
+import time
+import gc
+import os
+from pathlib import Path
+
 import asdf
 import numpy as np
-import sys
 from scipy.interpolate import interp1d
 import matplotlib.pyplot as plt
-import numpy.lib.recfunctions as rfn
-import time
-import gc
-import os
+import numba as nb
 
 from tools.merger import simple_load, get_slab_halo, extract_superslab
 
+# these are probably just for testing; should be removed for production
+DEFAULTS = {}
+DEFAULTS['sim_name'] = "AbacusSummit_highbase_c000_ph100"  # AbacusSummit_base_c000_ph006
+DEFAULTS['z_start'] = 0.3  # 0.8#0.5
+DEFAULTS['z_stop'] = 0.65  # 1.25#0.8#0.5
+
+
 # reorder in terms of their slab number
 def reorder_by_slab(fns):
-    i_sort = np.argsort(extract_superslab(fns))
-    tmp = fns.copy()
-    for i in range(len(fns)):
-        tmp[i] = fns[i_sort[i]]
-    return tmp
+    return sorted(fns, key=extract_superslab)
+
 
 # read redshifts from merger tree files
 def get_zs_from_headers(snap_names):
     zs = np.zeros(len(snap_names))
     for i in range(len(snap_names)):
         snap_name = snap_names[i]
-        f = asdf.open(snap_name,lazy_load=True, copy_arrays=True)
-        z = np.float(f.tree['header']['Redshift'])
-        f.close()
-        zs[i] = z
+        with asdf.open(snap_name) as f:
+            zs[i] = np.float(f["header"]["Redshift"])
```

Comment:
Don't need to say `f.tree['header']`; can just say `f['header']`.  I've made this change throughout the code.

Using `asdf.open()` as a context manager just saves typing `f.close()` later.

`lazy_load` is the default; `copy_arrays` doesn't really matter here since it's just the header.